shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1035,10 +1035,15 @@ - (void)appOrSceneBecameActive {
   if (_viewportMetrics.physical_width) {
     [self surfaceUpdated:YES];
   }
-  [self goToApplicationLifecycle:@"AppLifecycleState.resumed"];
+  [self performSelector:@selector(goToApplicationLifecycle:)
+             withObject:@"AppLifecycleState.resumed"
+             afterDelay:0.0f];
```

Comment:
This seems dangerous; what prevents us from mis-ordering lifecycle events because just one of them is delayed?

Reply:
> This seems dangerous; what prevents us from mis-ordering lifecycle events because just one of them is delayed?

When the device like iPhone14 pro is locked, the system mistakenly sends an "active" event, so we may need to cancel it if we receive an inactive notification in the same runloop.

Reply:
I understand what the goal of the PR is; that's not what I asked. I'm asking why we should have confidence that this will not mis-order the event relative to other lifestyle events in the cases where it's not cancelled.

Reply:
active/resumed is the final event of application lifecycle events when app changed from inactive to active.